test/xpu_api/random/interface_tests/distributions_methods_dp.pass.cpp

```diff
@@ -0,0 +1,210 @@
+// -*- C++ -*-
+//===-- distributions_methods_dp.pass.cpp -------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Testing of different distributions' methods with integer and double
+
+#include "support/utils.h"
+
+#if TEST_UNNAMED_LAMBDAS
+#include "common_for_distributions_methods.hpp"
+#endif // TEST_UNNAMED_LAMBDAS
+
+int
+main()
+{
+
+#if TEST_UNNAMED_LAMBDAS
+
+    sycl::queue queue = TestUtils::get_test_queue();
+    std::int32_t err = 0;
+
+    // Skip tests if DP is not supported
+    if (TestUtils::has_type_support<double>(queue.get_device())) {
+
+        std::cout << "---------------------------------------------------" << std::endl;
+        std::cout << "uniform_int_distribution<std::int32_t>" << std::endl;
+        std::cout << "---------------------------------------------------" << std::endl;
+        err += test<oneapi::dpl::uniform_int_distribution<std::int32_t>>(queue);
+#if TEST_LONG_RUN
+        err += test_vec<oneapi::dpl::uniform_int_distribution<sycl::vec<std::int32_t, 16>>>(queue);
```

Comment:
The thing is that doubles are used inside these distributions.
If I should rename this file how it will be more appropriate? 
`distributions_methods_int_dp_bool.pass.cpp`
Is it ok?

Reply:
I propose to keep it as is, assuming the dp implies the underlying implementation (we can add it as a comment in the file, that these types are tested, the implementation uses dp inside)